marlowe/src/Language/Marlowe/Scripts.hs

```diff
@@ -191,22 +191,18 @@ smallMarloweValidator
     -> [Input]
     -> ScriptContext
     -> Bool
-smallMarloweValidator MarloweParams{rolesCurrency, rolePayoutValidatorHash} MarloweData{..} inputs ctx@ScriptContext{scriptContextTxInfo} = do
-    let slotConfig = def :: TimeSlot.SlotConfig
+smallMarloweValidator MarloweParams{rolesCurrency, rolePayoutValidatorHash, slotConfig = (scSlotLength, scSlotZeroTime)} MarloweData{..} inputs ctx@ScriptContext{scriptContextTxInfo} = do
     let ownInput = case findOwnInput ctx of
             Just i -> i
             _      -> traceError "I0" {-"Can't find validation input"-}
     let scriptInValue = txOutValue $ txInInfoResolved ownInput
-    let (minTime, maxTime) =
-            case txInfoValidRange scriptContextTxInfo of
-                Interval.Interval (Interval.LowerBound (Interval.Finite l) True) (Interval.UpperBound (Interval.Finite h) False) -> (l, h)
+    let interval =
+            case TimeSlot.posixTimeRangeToContainedSlotRange TimeSlot.SlotConfig{..} $ txInfoValidRange scriptContextTxInfo of
+                -- FIXME restore this when mockchain implementation updates to correct one
```

Comment:
FYI, I relaxed the inclusiveness of the boundaries temporarily because the PAB seems to unpredictably choose the that inclusiveness when construction transactions.